src/builder-source-patch.c

```diff
@@ -186,6 +186,46 @@ builder_source_patch_download (BuilderSource  *source,
   return TRUE;
 }
 
+static gboolean
+git (GFile   *dir,
+     ...)
+{
+  gboolean res;
+  va_list ap;
+
+  va_start (ap, dir);
+  res = flatpak_spawn (dir, NULL, 0, NULL, "git", ap);
+  va_end (ap);
+
+  return res;
+}
+
+static gboolean
+ensure_git (GFile *dir)
+{
+  GFile *git_dir;
+  gboolean ret;
+  char *basename;
+
+  git_dir = g_file_get_child (dir, ".git");
+  ret = g_file_query_exists (git_dir, NULL);
+  g_object_unref (git_dir);
```

Comment:
I think this should try "git status" instead, because it might be run in a subdirectory of the directory with a .git in it.

Reply:
Sorry, don't understand. The patch would be applied to a sub-dir that's a git repo, not at the top-level?

Reply:
Say the toplevel is a git checkout and you want to apply a patch to a subdirectory only (dest=subdir in the source), then it will look for subdir/.git which doesn't exist, rather than the toplevel .git which does, and then wrongly do git init.